src/components/views/settings/Notifications.tsx

```diff
@@ -417,8 +418,10 @@ export default class Notifications extends React.PureComponent<IProps, IState> {
                 }
             }
             const kind = PushRuleKind.ContentSpecific;
+            const DOT = "DOT";
             for (const word of diff.added) {
-                await MatrixClientPeg.get().addPushRule('global', kind, word, {
+                const newKey = encodeURIComponent(word.replace(/\./g, DOT).trim());
```

Comment:
The spec says that only leading dots are not allowed. Can we just prefix the key with `custom-`? Ie: 
```suggestion
                const newKey = `custom-${word}`;
```

I'm also not clear on why we are URL encoding this as it should be safe/done for us by the http layers.